ui/src/router/components/form/components/ensembler_config/typeOptions.js

```diff
@@ -62,11 +75,10 @@ export const ensemblerTypeOptions = (engineProps) => {
     return typeOptions.filter((o) => o.value !== "standard");
   }
   // Ensembler must be selected when there is an experiment engine
-  const ensemblerOptions = typeOptions.filter((o) => o.value !== "nop");
-  if (
-    !engineProps?.standard_experiment_manager_config
-      ?.experiment_selection_enabled
-  ) {
+  const ensemblerOptions = typeOptions.filter(
+    (o) => o.value !== "nop" && (o.expenginetype === engineProps.type || o.expenginetype === undefined)
+  );
+  if (engineProps?.standard_experiment_manager_config?.experiment_selection_enabled === false) {
```

Comment:
This new condition (compared with `(!engineProps?.standard_experiment_manager_config?.experiment_selection_enabled)` previously) ensures that the experiment manager config has to explicitly disallow experiment selection before we remove the standard ensembler option (i.e. standard ensemblers will be selectable with custom experiment engines).

This contrasts with the previous condition which can return true whenever the `engineProps` does not have have the `standard_experiment_manager_config.experiment_selection_enabled` path defined (i.e. standard ensemblers are not selectable with custom experiment engines).